mne/simulation/simulation.py

```diff
@@ -0,0 +1,336 @@
+# author: ngayraud.
+#
+# Created on Tue Feb 20 11:14:54 2018.
+
+import numpy as np
+
+from .source import simulate_sparse_stc
+from ..forward import apply_forward_raw
+from ..utils import warn, logger
+from ..io import RawArray
+from ..io.constants import FIFF
+
+from .waveforms import get_waveform
+from .noise import generate_noise_data
+
+
+class Simulation(dict):
+    """Simulation of meg/eeg data.
+
+    Parameters
+    ----------
+    fwd : Forward
+        a forward solution containing an instance of Info and src
+    n_dipoles : int
+        Number of dipoles to simulate.
+    labels : None | list of Labels
+        The labels. The default is None, otherwise its size must be n_dipoles.
+    location : str
+        The label location to choose a dipole from. Can be ``random`` (default)
+        or ``center`` to use :func:`mne.Label.center_of_mass`. Note that for
+        ``center`` mode the label values are used as weights.
+    subject : string | None
+        The subject the label is defined for.
+        Only used with location=``center``.
+    subjects_dir : str, or None
+        Path to the SUBJECTS_DIR. If None, the path is obtained by using the
+        environment variable SUBJECTS_DIR. Only used with location=``center``.
+    waveform: list of callables/str of length n_dipoles | str | callable
+        To simulate a waveform (activity) on each dipole. If it is a string or
+        a callable, the same activity will be generated over all dipoles
+    window_times : array | list | str
+        time window(s) to generate activity. If list, its size should be
+        len(waveform). If str, should be ``all`` (default)
+
+    Notes
+    -----
+    Some notes.
+    """
+
+    def __init__(self, fwd, n_dipoles=2, labels=None, location='random',
+                 subject=None, subjects_dir=None, waveform='sin',
+                 window_times='all'):
+        self.fwd = fwd  # TODO: check fwd
+        labels, n_dipoles = self._get_sources(labels, n_dipoles)
+        self.update(n_dipoles=n_dipoles, labels=labels, subject=subject,
+                    subjects_dir=subjects_dir, location=location,
+                    info=self.fwd['info'])
+        self['info']['projs'] = []
+        self['info']['bads'] = []
+        self['info']['sfreq'] = None
+        self.waveforms = self._get_waveform(waveform)
+        self.window_times = self._get_window_times(window_times)
```

Comment:
I don't fully get the pattern here. The object has no public method and does everything in the init. Isn't all this doable with a plain function?